src/components/v1/V1Project/modals/DownloadPaymentsModal.tsx

```diff
@@ -0,0 +1,117 @@
+import { t, Trans } from '@lingui/macro'
+import { Modal } from 'antd'
+import InputAccessoryButton from 'components/shared/InputAccessoryButton'
+import FormattedNumberInput from 'components/shared/inputs/FormattedNumberInput'
+import { readProvider } from 'constants/readProvider'
+import { V1ProjectContext } from 'contexts/v1/projectContext'
+import { useCallback, useContext, useEffect, useState } from 'react'
+import { fromWad } from 'utils/formatNumber'
+import { querySubgraphExhaustive } from 'utils/graph'
+
+export default function DownloadPaymentsModal({
+  visible,
+  onCancel,
+}: {
+  visible: boolean | undefined
+  onCancel: VoidFunction | undefined
+}) {
+  const [latestBlockNumber, setLatestBlockNumber] = useState<number>()
+  const [blockNumber, setBlockNumber] = useState<number>()
+  const [loading, setLoading] = useState<boolean>()
+  const { projectId, tokenSymbol, handle } = useContext(V1ProjectContext)
+
+  useEffect(() => {
+    readProvider.getBlockNumber().then(val => {
+      setLatestBlockNumber(val)
+      setBlockNumber(val)
+    })
+  }, [])
+
+  const download = useCallback(async () => {
+    if (blockNumber === undefined || !projectId) return
+
+    setLoading(true)
+
+    const rows = [
+      [t`Amount paid`, t`Date`, t`Payer`, t`Beneficiary`, t`Note`], // CSV header row
+    ]
+
+    try {
+      const payments = await querySubgraphExhaustive({
+        entity: 'payEvent',
+        keys: ['caller', 'beneficiary', 'amount', 'timestamp', 'note'],
+        orderBy: 'timestamp',
+        orderDirection: 'desc',
+        block: {
+          number: blockNumber,
+        },
+        where: {
+          key: 'project',
+          value: projectId.toString(),
+        },
+      })
+
+      if (!payments) throw new Error('No data.')
```

Comment:
**suggestion** is it worth displaying some error message in the UI?

Reply:
Yeah fair point. Especially since we've been seeing more Graph errors lately than usual.

In general we could probably be relying more on the Antd `Notification` element, I'll whip something up

Reply:
Ohhh yeah, great idea ❤️ .